src/views/studio/MyVideosView/MyVideosView.tsx

```diff
@@ -0,0 +1,189 @@
+    progress: undefined,
+  }))
+  const videosWPlaceholders = [...(videos || []), ...placeholderItems]
+  const handleOnResizeGrid = (sizes: number[]) => setVideosPerRow(sizes.length)
+
+  const FetchMoreVideos = (page: number) =>
+    fetchMore({
+      variables: {
+        // substract 1 coz offset index starts at 0
+        offset: videosPerPage * (page - 1),
+        limit: videosPerPage,
+      },
+    })
+  // console.log({ videos, totalCount, hasAnyVideos, isLoading, isPublic_eq })
+  return (
+    <ViewContainer>
+      <StyledText variant="h2">My Videos</StyledText>
+
+      {hasAnyVideos === false ? (
+        <AddVideoView />
+      ) : (
+        <>
+          <TabsContainer>
+            <Tabs initialIndex={0} tabs={[...TABS]} onSelectTab={setCurrentTab} />
+          </TabsContainer>
+          {isDraftTab && (
+            // Should this really be dismissable?
+            <StyledDismissibleMessage
+              id="video-draft-saved-locally-warning"
+              title={'Video Drafts are saved locally'}
+              description={
+                'This mean you can only access one on the device you used to create it. Clearing your browser history will delete all your drafts.'
+              }
+            />
+          )}
+          <Grid maxColumns={null} onResize={handleOnResizeGrid}>
+            {!isDraftTab &&
+              videosWPlaceholders
+                // this makes for a smoother transition between pages
+                .slice(0, videosPerPage)
+                .map((video, idx) => (
+                  <VideoPreviewPublisher
+                    key={idx + '-' + currentTabName + '-' + currentPage}
+                    id={video.id}
+                    showChannel={false}
+                    isLoading={loading}
+                  />
+                ))}
+            {isDraftTab &&
+              drafts
+                // pagination slice
+                .slice(videosPerPage * (currentPage - 1), (currentPage - 1) * videosPerPage + videosPerPage)
+                .map((draft, idx) => (
+                  <VideoPreviewPublisher
+                    key={idx + '-' + currentTabName + '-' + currentPage}
+                    id={draft.id}
+                    showChannel={false}
+                    isDraft
+                    onEditVideoClick={() => ({})}
+                    onDeleteVideoClick={() => {
+                      removeDraft(draft.id)
+                    }}
+                  />
+                ))}
+          </Grid>
+          {((isDraftTab && drafts.length === 0) || (!isDraftTab && totalCount === 0 && !isLoading)) && <AddVideo />}
+          <PaginationContainer>
+            <Pagination
+              onChangePage={(page) => {
+                setCurrentPage(page)
+                currentTabName !== 'Drafts' && FetchMoreVideos(page)
+              }}
+              onMouseEnterPage={(page) => {
+                currentTabName !== 'Drafts' && FetchMoreVideos(page)
+              }}
```

Comment:
I see you tried prefetching here but I'm not sure if we should do that right now. I agree it's a good user experience and something we should have but it would need to be done properly and I don't think it will be this simple. To keep this PR contained, I suggest we keep it out.

https://user-images.githubusercontent.com/12646744/111625231-e8a39b80-87ec-11eb-939e-e7b6fa6c734b.mov

Reply:
(take a look at the request count in bottom right)

Reply:
Yes I tried solving this but it would need a memorized throttle fn I think, yes def tricky